### ISSUE TITLE
Exceptions bubble up with unhelpful "undefined method `join' for String" message

### ISSUE THREAD

**Issue by lencioni** [tdooner/mosaic#17]
![screen shot 2015-05-23 at 1 54 53 pm](https://cloud.githubusercontent.com/assets/195534/7785748/5cb9d812-0153-11e5-812a-cd0cb51e104f.png)

**Comment by tdooner** [comment]
Yeah, this needs to be better. This happens any time the application raises an exception. Ideally, we should install middleware which can catch that exception and provide a good user-facing error message.

**Comment by lencioni** [comment]
Sure, but I'm more interested in the dropbox download link actually working.

**Comment by tdooner** [comment]
Roger that. I tried it and it just seemed to work, so I suspect it has to do with the file moving around or something. I opened #20 for it.
